projects/storefrontlib/src/cms-components/product/product-list/product-facet-navigation/services/facet.service.ts

```diff
@@ -59,15 +61,15 @@ export class FacetService {
    * If the optional value argument is provided the expanded state will be set
    * to this value, regardless of the current `expanded` state.
    */
-  toggleExpand(facet: Facet, value?: boolean) {
+  toggle(facet: Facet, isExpanded: boolean): void {
     const state = this.getStateSnapshot(facet);
 
     const toggledState = {
-      toggled: value ?? !state.toggled,
+      toggled: isExpanded ? ToggleState.COLLAPSED : ToggleState.EXPANDED,
     } as FacetCollapseState;
 
-    if (!state.maxVisible) {
```

Comment:
Question: Why we've dropped checking for `state.maxVisible` now?

Reply:
This was buggy actually. The maxVisible is always set during initialisation. Only when we collapse the facet group, we like to reinitialise the max visible values.